tests/xbot/common/logic/LatchTest.java

```diff
@@ -0,0 +1,77 @@
+package xbot.common.logic;
+
+import org.junit.Test;
+
+import java.util.Observable;
+import java.util.Observer;
+
+import static junit.framework.TestCase.assertEquals;
+
+public class LatchTest {
+
+    @Test
```

Comment:
You don't have any tests that set the same value more than once, so you'll never exercise one branch of the "isDifferent" code in latch.

(It might be a good idea to go to Help -> Eclipse Marketplace and install EclEmma, a code coverage tool, and you can see what code never gets executed)